packages/framer-motion/src/projection/node/create-projection-node.ts

```diff
@@ -100,6 +101,26 @@ function resetDistortingTransform(
     }
 }
 
+function isOptimisedAppearTree(projectionNode: IProjectionNode) {
+    projectionNode.hasCheckedOptimisedAppear = true
+    if (projectionNode.root === projectionNode) return false
+
+    const { visualElement } = projectionNode.options
+
+    if (!visualElement) {
+        return false
+    } else if (getOptimisedAppearId(visualElement)) {
+        return true
+    } else if (
+        projectionNode.parent &&
+        !projectionNode.parent.hasCheckedOptimisedAppear
+    ) {
+        return isOptimisedAppearTree(projectionNode.parent)
+    } else {
+        return false
+    }
```

Comment:
nit: since it's all return statements, maybe we can get rid of the else blocks 
```suggestion
    if (!visualElement) return false  
    if (getOptimisedAppearId(visualElement)) return true
    if (
        projectionNode.parent &&
        !projectionNode.parent.hasCheckedOptimisedAppear
    ) return isOptimisedAppearTree(projectionNode.parent)

    return false
```